### PR TITLE
update SMS verification

### LINKED ISSUE #3514: SMS process failed: all gas used up
I went thru the process on testnet: https://testnet.etherscan.io/tx/0x6ed045aeaaa8d0815d44d98e534655874969dfa144bfefa97f8e80729bce2c28

It failed after I signed the tx that sends back the puzzle magic sentence.

`Failed to send the verification code: Transaction failed, all gas used up.`

### PR DESCRIPTION
**Pull request by derhuerst**
This PR adapts the SMS verification modal to the [changes in the server](https://github.com/ethcore/sms-verification/commits/0b57193f37de3dee48be48f311cd008a9d164610) and adds a few more checks. Fixes #3514.

@chevdor Please check, I will keep the `ropsten` server running now.

22e1384 and c62c72d make the modal wait until the `Puzzled` event, posted by the server, has actually been mined. This prevents the follow-up `confirm` tx from failing (and therefore using up all gas).

As requested by @gavofyork, 6509252 makes the modal check if the code is valid using `contract.confirm.call()`.

### CODE REVIEW

**Comment by chevdor** [comment]
I cannot check right now but I will within 2h max.

**Comment by derhuerst** [comment]
@jacogr @ngotchac 

`util/subscribe-to-event` is a helper function that returns an event-based interface to listen for events.

Because the promise returned by `Contract.prototype.subscribe` resolves *after* the callback has been called for the first time, it is hard to unsubscribe if you already got what you want. You will have to store that you want to unsubscribe as soon as the promise resolves.

I think it now depends which interface is easier to consume. If you want many events, using the event-based interface is really nice. For one, it won't be worth the boilerplate.

**Comment by chevdor** [comment]
I merged `jr-update-sms-verification` and rebuilt.
I am still getting `Failed to request a confirmation SMS: Failed to fetch`

**Comment by derhuerst** [comment]
@chevdor Sorry, I forgot to mention that it only works on `ropsten`. For me and @jacogr, it just worked.

**Comment by chevdor** [comment]
@derhuerst I also forgot to mention it I am also (now) on Ropsten. If it works for you both, I can propose you to connect with me and we check together or we leave it as it is, and I test again once the PRs is merged. Up to you.

**Comment by derhuerst** [comment]
Still to do:

- [ ] show the server errors in the UI

File: js/src/util/is-testnet.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+export default (chain) => {
+  return chain === 'morden' || chain === 'ropsten' || chain === 'testnet';
```
Comment (jacogr) [inline js/src/util/is-testnet.js]: ? Redux has isTest, don't want to maintain it in yet another place, we just simplified and got it down to 1. (And had to make changes.)

In addition, we are really starting to clutter /utils now with all these single function files. There is no reason why things can't be grouped, e.g. had to move nullable to protypes once we added another one. (General grumble)

File: js/src/util/subscribe-to-event.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import EventEmitter from 'eventemitter3';
```
Comment (jacogr) [inline js/src/util/subscribe-to-event.js]: Would have really preferred a general solution that (a) doesn't add another way of doing things, (b) is available to everybody. But suppose it solves the immediate issue, so once we have a general solution in-place will have to come back.

File: js/src/util/is-testnet.js (hunk 1 above)
Comment (derhuerst) [inline js/src/util/is-testnet.js]: Fixed as of 88cb1df92f063b53a53bb1846e44c9dc0e3abd68.

**Pull request merged**